.github/workflows/lint.yaml

```diff
@@ -0,0 +1,50 @@
+name: lint
+
+on:
+  pull_request:
+    types: [opened, edited, reopened, synchronize]
+    branches:
+      - main
+      - feature/*
+  push:
+    branches:
+      - main
+      - feature/*
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name }}-${{ inputs.platform }} @ ${{ github.event.label.name || github.event.pull_request.number || github.sha }} @ ${{ github.event.label.name && github.event.pull_request.number || github.event.action }}
+  cancel-in-progress: true
+
+permissions: {}
+
+jobs:
+  lint:
+    name: Pre-Commit
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - name: Install pre-commit
+        run: |
+          sudo apt update
+          sudo apt install python3 gn pipx
+          pipx install pre-commit
```

Comment:
Why do we recommend people install pre-commit via apt but install it in this action via pip? If those end up out of sync won't we run into issues?

Reply:
Correct me if I'm wrong, but if we use pip installs for both the container and our local workstations, won't that keep versions in sync?

Reply:
Also, shouldn't the installation be part of the docker image instead of doing it in the action?

Reply:
We can switch to the pipx install for our local workstations to keep them in sync.

Previously with our lint action in Cobalt we also installed dependencies and pre-commit in the action rather than in the docker image. According to Kaido both installation methods work and this way skips one layer of abstraction.